src/app/modules/group/components/pending-request/pending-request.component.ts

```diff
@@ -49,15 +51,78 @@ export class PendingRequestComponent implements OnInit, OnChanges {
   currentSort: string[] = [];
   includeSubgroup = false;
   collapsed = true;
-  status: 'loading' | 'loaded' | 'empty' |'error' = 'loading';
+
+  state: 'fetching' | 'ready' |'error' = 'fetching';
 
   ongoingActivity: Activity = 'none';
 
+  private dataFetching = new Subject<{ groupId?: string, includeSubgroup: boolean, sort: string[] }>();
+
+  private dataUpdating = new Subject<{ action: Action, requests: PendingRequest[] }>();
+
   constructor(
     private getRequestsService: GetRequestsService,
     private requestActionService: RequestActionsService,
     private messageService: MessageService
-  ) {}
+  ) {
+    this.dataFetching.pipe(
+      switchMap(params =>
+        merge(
+          of(fetchingState()),
+          this.getRequestsService.getPendingRequests(params.groupId, params.includeSubgroup, params.sort).pipe(map(readyState))
+        )
+      )
+    ).subscribe(
+      state => {
+        this.state = state.tag;
+        if (isReady(state)) {
+          this.requests = state.data;
+          if (this.requests.length > 0) this.collapsed = false;
+        }
+      },
+      _err => {
+        this.state = 'error';
+      }
+    );
+
+    this.dataUpdating.pipe(
+      switchMap(params =>
```

Comment:
Actually actions must not be switched-mapped... they cannot be cancelled or half-done if another one arrive (even if it is not really possible). 
The remark is valid for any such actions (in other components as well).

Reply:
fixed